game/src/lib.rs

```diff
@@ -254,10 +255,19 @@ fn finish_app_setup(
 
     let states: Vec<Box<dyn State<App>>> = if title {
         vec![Box::new(TitleScreen::new(ctx, app))]
-    } else {
-        let mode = maybe_mode
-            .unwrap_or_else(|| GameplayMode::Freeform(app.primary.map.get_name().clone()));
+    } else if let Some(mode) = maybe_mode {
         vec![SandboxMode::simple_new(ctx, app, mode)]
+    } else {
+        // We got here by just passing --dev and a map as flags; we're just looking at an empty
+        // map. Start in the daytime.
+        vec![
+            SandboxMode::simple_new(
+                ctx,
+                app,
+                GameplayMode::Freeform(app.primary.map.get_name().clone()),
+            ),
+            TimeWarpScreen::new(ctx, app, Time::START_OF_DAY + Duration::hours(6), None),
```

Comment:
Is this too heavy-handed a way of advancing sim time? Usually the map is empty, so the only simulation things actually happening are traffic lights changing. When we have more buses spawning, they'd also be simulated. Since advancing 6 hours on a mostly empty map isn't instantaneous, the time warp screen seems better than just silently blocking.

Reply:
From a UX standpoint, it's awkward that we complete one loading screen only to immediately pop up a second loading screen (the automatic "jump to time" modal is effectively a loading screen).

Did you consider showing some kind of output on the primary loading screen so that we can skip the "jump to time" modal?

Maybe because of the way we're doing threading that's not feasible/easy, in which case this is fine for now, but I think we should eventually address it.

Reply:
Not sure if this is what you meant, but now we're using a "normal" loading screen instead of the jump-to-time modal. Because of how loading screens try to skip drawing, I haven't actually spotted this one yet, even on larger maps. But I think this is an improvement over jump-to-time because it doesn't draw or otherwise use `time_limited_step` and get throttled by the widgetry frame-rate. Overall feels faster.